### PR TITLE
[LCI] Implement Intrepid Paleontologist

### PR DESCRIPTION
**Pull request by ssk97**
Also fixes a bug that I introduced during the Watcher deep copy code changes where the copied watcher would lose track of the original permanent. Since the actual source of the effect doesn't matter, I'm making the SpellAbility of the cast spell be the source rather than the actual original permanent (which can't easily be tracked by the Watcher)

I *think* this has the correct interaction with copy effects (do not add a counter to the copy since it wasn't cast) but I didn't check with a judge.

### CODE REVIEW

**Review by xenohedron — COMMENTED** [review]
all else looks good

File: Mage.Tests/src/test/java/org/mage/test/cards/watchers/HallOfTheBanditLordTest.java [hunk 1]
```diff
@@ -50,4 +50,30 @@ public void testNotGainsHaste() {
 
         this.assertAbility(playerA, "Ember Hauler", HasteAbility.getInstance(), false);
     }
+    //Test that a copied Hall of the Bandit Lord works properly
```
[the remark this answers is not in the document]
Comment (ssk97) [reply Mage.Tests/src/test/java/org/mage/test/cards/watchers/HallOfTheBanditLordTest.java:53]: I suppose it wouldn't really hurt beyond a tiny amount of extra time running tests, but I can't think of any errors that could catch - there's no way for it to result in a crashing null anymore like the quick fix I originally did to get rid of the infinite loop during the deep copy code changes.

[the remark this answers is not in the document]
Comment (xenohedron) [reply Mage.Tests/src/test/java/org/mage/test/cards/watchers/HallOfTheBanditLordTest.java:53]: fine as is then

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/h/HallOfTheBanditLord.java [hunk 2]
```diff
@@ -100,7 +97,8 @@ public void watch(GameEvent event, Game game) {
             if (creatures.contains(event.getSourceId())) {
                 ContinuousEffect effect = new GainAbilityTargetEffect(HasteAbility.getInstance(), Duration.Custom);
                 effect.setTargetPointer(new FixedTarget(event.getSourceId(), game));
-                game.addEffect(effect, game.getAbility(sourceAbilityID, this.getSourceId()).orElse(null)); //null will cause an immediate crash
+                Ability source = game.getPermanent(event.getSourceId()).getSpellAbility();
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/h/HallOfTheBanditLord.java:100]: Looks like this caused NPE in #12162
